volatility3/framework/plugins/windows/netstat.py

```diff
@@ -188,16 +206,30 @@ def enumerate_structures_by_port(
             )
             yield curr_obj
 
+            try:
+                next_obj_address = cls._decode_pointer(curr_obj.Next)
+            except exceptions.InvalidAddressException:
+                return
+
             # if the same port is used on different interfaces multiple objects are created
             # those can be found by following the pointer within the object's `Next` field until it is empty
-            while curr_obj.Next:
-                curr_obj = context.object(
-                    obj_name,
-                    layer_name=layer_name,
-                    offset=cls._decode_pointer(curr_obj.Next) - ptr_offset,
-                )
+            while next_obj_address:
+                try:
+                    curr_obj = context.object(
+                        obj_name,
+                        layer_name=layer_name,
+                        offset=next_obj_address - ptr_offset,
+                    )
+                except exceptions.InvalidAddressException:
+                    return
+
                 yield curr_obj
 
+                try:
+                    next_obj_address = cls._decode_pointer(curr_obj.Next)
```

Comment:
Subtle difference between these.  In the old one we checked whether the `curr_obj.Next` was false, in this one we check if `cls._decode_pointer(curr_obj.Next)` is false.  Are these both likely to trip on the same values?

Reply:
It would work the same. If `Next` points to a value of 0 then the `decode_ptr` would do:

```
value = value & 0xFFFFFFFFFFFFFFFC

return value
```

With 0 as the starting input for `value`.

Reply:
versus if `Next` is an invalid address then the exception handler would trigger